Container Analysis samples

DanSanche opened the pull request:
added Python samples for the [Container Analysis](https://cloud.google.com/container-registry/docs/container-analysis) GA

leahecole commented:
Disclaimer: I have NO container_analysis experience, but have used this repo before :) Here's how running this went:

-Pretty please add a README so users know how you want them to run your code (that they need to install the requirements, let them know what needs to be enabled in their GCP Project, etc., any environment vars). Given that I've never used container analysis, I failed a bunch before I was able to successfully run stuff
- When running pip install -r requirements.txt in a virtual environment on your branch, I run into this dependency error (although when I pulled the upstream into my clone of your repo and rebased your work on top of it, it was okay) 
```
ERROR: grafeas 0.1.0 has requirement google-api-core[grpc]<2.0.0dev,>=1.6.0, but you'll have google-api-core 0.1.4 which is incompatible.
ERROR: google-cloud-containeranalysis 0.1.0 has requirement google-api-core[grpc]<2.0.0dev,>=1.4.1, but you'll have google-api-core 0.1.4 which is incompatible.
```
- the test_pubsub failed for me with a 404 resource not found resource=container-analysis-occurrences-v1beta1 - this is probably on me, and a README may have helped me not mess this up, but in case it's not me, double check that this test can run independently of the others

DanSanche commented:
@leahecole I added a README, let me know if that looks better. Your pubsub issues may have been because the team recently switched from the beta topic to the GA one, so I updated the sample to use the new topic. Let me know what you think

leahecole commented on container_registry/container_analysis/README.md:
````diff
@@ -0,0 +1,59 @@
+
+```
+    $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples
+    $ cd python-docs-samples/container-registry/container-analysis
````
Should be `cd python-docs-samples/container_registry/container_analysis/`

(underscores, not hyphens)

DanSanche replied:
Wow good catch! I copy/pasted from the ruby README, must have missed that

leahecole commented on container_registry/container_analysis/README.md:
````diff
@@ -0,0 +1,59 @@
+```
+    $ python3 -m venv ./venv
+    $ source ./venv/bin/activate
+    $ pip install -r requirements.txt
````
Got two errors installing dependencies (note, I did fetch the upstream and rebase your work on top before I tried anything)

`grafeas 0.1.0 has requirement google-api-core[grpc]<2.0.0dev,>=1.6.0, but you'll have google-api-core 0.1.4 which is incompatible.
google-cloud-containeranalysis 0.1.0 has requirement google-api-core[grpc]<2.0.0dev,>=1.4.1, but you'll have google-api-core 0.1.4 which is incompatible.`

DanSanche replied:
Hmm I haven't been able to reproduce this. And you're using a virtualenv?

It's strange because I'm seeing google_api_core-0.1.4 in my site-packages, but I didn't get any error messages

DanSanche replied:
ok, actually I updated the pubsub version, and I think everything should be ok now. (that updated google_api_core to be in the right range)

leahecole commented on container_registry/container_analysis/README.md:
````diff
@@ -0,0 +1,59 @@
+    $ export GOOGLE_CLOUD_PROJECT="YOUR_PROJECT_ID"
+```
+
+1. **Run Tiests**
````
Tiests -> Tests

leahecole commented on container_registry/container_analysis/README.md:
````diff
@@ -0,0 +1,59 @@
+1. **Set Environment Variables**
+
+```
+    $ export GOOGLE_CLOUD_PROJECT="YOUR_PROJECT_ID"
````
GOOGLE_CLOUD_PROJECT -> GCLOUD_PROJECT

leahecole replied:
Should users also set their application credentials here?

DanSanche replied:
Is that required? I don't have it set in my terminal and it's working, I assume because I authenticated locally with "gcloud login"

I'm not really sure how much detail is required for set up here, so I copied this readme from a different project

DanSanche commented:
Note that the tests aren't actually running, since this PR is based on a fork. I forgot that quirk of this repo

Maybe after we resolve these comments I can close this and open another one as an internal branch? Let me know what you think

leahecole commented:
Ooo I forgot that's a quirk of this repo also. @DanSanche I think that after resolution, opening another one and linking it here in the comments is a fabulous idea.

DanSanche commented:
@leahecole I believe I resolved the issues you mentioned, except setting application credentials. Let me know if you think that should be added

leahecole commented:
Sorry, didn't process that properly. I think adding the application creds is a good idea because right now it throws a warning if you don't

DanSanche commented:
sounds good, I'll add it to the new PR
